### ISSUE TITLE
MCP reports are blank

### ISSUE THREAD

**Issue by manz111** [Adobe-Consulting-Services/acs-aem-commons#3140]
## Required Information

* [ ] AEM Version, including Service Packs, Cumulative Fix Packs, etc: AEM Could
* [ ] ACS AEM Commons Version: latest
* [ ] Reproducible on Latest? yes

### Expected Behavior

MCP reports to be generated after running the MCP process

### Actual Behavior

We are unable to see any MCP reports in the console on AEM cloud. Below is the screenshot of the same -

![95c53f49-f69d-458b-932f-a248f653ec6d](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/139532233/24359329-19db-4aba-97bf-b78c8fcb390f)

### Steps to Reproduce

1. Start an MCP process
2. Once complete, no reports are seen in the console
3. But reports are getting stored under the node - var/acs-commons/mcp/instances

![56761890-71f4-4df5-9846-e17ad0fea595](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/139532233/8607d6ef-e33d-47d1-9b47-c71952def67f)

Upon double clicking on the report node in CRX, we see the below data :

![909b5fce-6100-4be1-b979-f1f68f0f6d77](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/139532233/7daff003-5b0c-4b5b-b9d6-cd1713c3057a)

![a8480707-031e-436a-bf4c-4f5c57dda2d5](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/139532233/278165fd-c576-4f05-bcba-472f70b7b4e5)

We are not seeing any errors getting logged wrt MCP reports.

### Links

https://author-p57322-e572072.adobeaemcloud.com/apps/acs-commons/content/manage-controlled-processes.html

**Comment by davidjgonzalez** [comment]
@manz111 did this just start happening? Do you see any errors in the browsers JS console?

**Comment by davidjgonzalez** [comment]
![2023-07-14 at 4 54 PM](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/1451868/ed00eaef-9bf5-44e2-8a8a-01b5afac8354)

This is on latest ACS AEM Commons and Adobe Experience Manager 2023.6.12255.20230608T053118Z-230601

These seem to show up here ... can you check your error logs and your JS console? This might be something strange going on w your env.

**Comment by manz111** [comment]
Yes @davidjgonzalez , Please find below the screenshot of the error in console log - 
<img width="956" alt="image" src="https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/139532233/cd803ee7-bfad-4d92-80f3-52d3d487b29f">

**Comment by davidjgonzalez** [comment]
Are there other errors if you dont filter? Also - are you seeing any errors (404, 403, 500) in the Network tab?

**Comment by manz111** [comment]
No, I don't see any 40x errors in the network tab. Adding screenshot below - 
![image](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/139532233/d2df7080-9be7-44f7-869a-dc347e64a7c4)

**Comment by YegorKozlov** [comment]
@manz111 can you package `/var/acs-commons/mcp` on your instance and attach to this ticket? I'd like to see which instance is causing the NPE.

**Comment by MANID048** [comment]
Is this issue resolved?

**Comment by davidjgonzalez** [comment]
@manz111 any chance you could check if the following build helps resolve this (jar is the only thing w/ changes but attached all package too). It sounds like the data under /var/acs-commons/mcp is incomplete .. so this should slot the items w/ the missing starttimes to the bottom of the dashboard so you can deal w them via the UI. That said, its hard to reproduce since we dont know what your `/var/acs-commons/mcp` looks like (would still be great if you could package that up and share it -- you could send direct to dgonzale@adobe.com if it has data you dont want out in the open).


[acs-aem-commons-bundle-6.0.15-SNAPSHOT.jar.txt](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/12261119/acs-aem-commons-bundle-6.0.15-SNAPSHOT.jar.txt)
[acs-aem-commons-all-6.0.15-SNAPSHOT.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/12261120/acs-aem-commons-all-6.0.15-SNAPSHOT.zip)

**Comment by davidjgonzalez** [comment]
@MANID048 are you having this problem as well?

**Comment by manz111** [comment]
Hi @davidjgonzalez , I have shared with you the /var/acs-commons/mcp package from our instance to you email (sent from manasa.p2@macys.com). Waiting to hear back from you.

**Comment by davidjgonzalez** [comment]
Thanks @manz111  - i received the zip.

It looked like there are 6 invalid processes (the mcp process instances are missing almost all data) which was breaking the report. I've put a fix in to atleast display these so you can investigate (and it wont break the report UI totally) in https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/3154

![2023-08-07 at 10 30 AM](https://github.com/Adobe-Consulting-Services/acs-aem-commons/assets/1451868/a2708f30-4cc8-4f8b-a90c-542abfdb81d6)

I can't tell what sort of process generated these null process nodes (they're for the most part empty) - do you have any sense if a specific process was run when these were created? While waiting for this release, you can review/delete those nodes manually:


Invalid (5563E7F940240A0F) | Archived process at /var/acs-commons/mcp/instances/5563E7F940240A0F is null | n/a | n/a | ??? |  
-- | -- | -- | -- | -- | --
Invalid (0C7569E69383E5D4) | Archived process at /var/acs-commons/mcp/instances/0C7569E69383E5D4 is null | n/a | n/a | ??? |  
Invalid (568FFE4CD9A4012C) | Archived process at /var/acs-commons/mcp/instances/568FFE4CD9A4012C is null | n/a | n/a | ??? |  
Invalid (21CB8D45D0E8F7EB) | Archived process at /var/acs-commons/mcp/instances/21CB8D45D0E8F7EB is null | n/a | n/a | ??? |  
Invalid (7899AA7633063FE3) | Archived process at /var/acs-commons/mcp/instances/7899AA7633063FE3 is null | n/a | n/a | ??? |  
Invalid (0B29982E13D3C8F9) | Archived process at /var/acs-commons/mcp/instances/0B29982E13D3C8F9 is null | n/a | n/a | ??? |

**Comment by manz111** [comment]
Thank you @davidjgonzalez for helping me in debugging. Deleted these nodes and the MCP reports have started to render. But could you let me know how you were able to find out the invalid nodes? Was it through a query or something?

**Comment by davidjgonzalez** [comment]
It was via the acs commons patch in this ticket. I implemented better handling of invalid results to be able to see the results. :)
